### PR TITLE
Initial content

### PR DESCRIPTION
**Pull request by jamiecoleman92**
This includes, README's, Scripts, Dockerfile's and a test for the source2image tool.

### CODE REVIEW

File: source2image/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+
+It is possible to use the Source2Image tool to build your source code into a WebSphere Liberty Docker image. This repository was created to store the resources needed to use this feature. Instructions to achieve this are available on WASDev via the following link:
+
+
+Explanation of Dockerfile:
+
+The first line "FROM" pulls down the WebSphere Liberty image from Dockerhub.
+
+The next set of "RUN" commands install maven, create a directory to store the source code and install a jdk which is needed for OpenShift.
```
Comment (davidcurrie) [inline source2image/README.md]: Capitalize JDK

File: README.md [hunk 2]
```diff
@@ -1,2 +1,4 @@
 # ci.docker.openshift
-Documentation and assets for OpenShift and WebSphere Liberty.
+Documentation and assets for using OpenShift v3 with WebSphere Liberty.
+More information regarding how to deploy OpenShift with WebSphere Liberty can be found on WASDev at
```
Comment (davidcurrie) [inline README.md]: Create this as a hyperlink e.g. "can be found on [WASDev](https://developer.ibm.com/wasdev/docs/running-websphere-liberty-openshift-v3/)".

File: source2image/dockerfile [hunk 3]
```diff
@@ -0,0 +1,42 @@
+FROM websphere-liberty:javaee7
```
Comment (davidcurrie) [inline source2image/dockerfile]: Filename appears to be "dockerfile" - change to "Dockerfile"

File: source2image/README.md (hunk 1 above)
Comment (davidcurrie) [inline source2image/README.md]: Actually, change it to "and install an SDK which is needed to build the application"

File: source2image/usage [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# inform the user how to use the image
+cat <<EOF
+This is an S2I WebSphere Liberty base image that is responsible for consuming a JEE project and deploying it onto a WebSphere Liberty Application server.
```
Comment (davidcurrie) [inline source2image/usage]: lowercase "a" for "application server"

File: source2image/usage [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+To Use it, install S2I at: https://github.com/openshift/source-to-image.
```
Comment (davidcurrie) [inline source2image/usage]: lowercase "u" for "use"

Comment (davidcurrie) [inline source2image/usage]: "install S2I from"

File: source2image/usage [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+Sample invocation:
+
+s2i build https://github.com/WASdev/sample.ferret.git websphere-liberty libertys2itest
+
+You can then run the resulting image via
+docker run -p <image name>
```
Comment (davidcurrie) [inline source2image/usage]: either use libertys2itest everywhere or <image name> everywhere

File: source2image/usage [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+Details:
+If a pom.xml file is present in the source directory then the project is built with maven and the output in target will be copied into the Liberty deployments directory /opt/ibm/wlp/usr/servers/defaultServer/dropins/.
+The WebSphere-Liberty start script located in /opt/ibm/wlp/bin/ is called when the container is run which in turn checks to see if an application is present in the /dropins folder then runs it.
```
Comment (davidcurrie) [inline source2image/usage]: remove the hyphen in WebSphere-Liberty

File: source2image/usage [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+Sample invocation:
+
+s2i build https://github.com/WASdev/sample.ferret.git websphere-liberty libertys2itest
+
+You can then run the resulting image via
+docker run -p <image name>
+
+Details:
+If a pom.xml file is present in the source directory then the project is built with maven and the output in target will be copied into the Liberty deployments directory /opt/ibm/wlp/usr/servers/defaultServer/dropins/.
```
Comment (davidcurrie) [inline source2image/usage]: What if a pom.xml file is not present?

File: source2image/dockerfile [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+# Install Maven
```
Comment (davidcurrie) [inline source2image/dockerfile]: I'd actually install maven here and separate it out from the SDK install

File: source2image/usage (hunk 8 above)
Comment (davidcurrie) [inline source2image/usage]: /config/dropins

File: source2image/assemble [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+  if [ $ERR -ne 0 ]; then
+    echo "Aborting due to error code $ERR from mvn package"
+    exit $ERR
+  fi
+
+
+#Copy .war file from the source directory
+  echo "Copying built war files into $DEPLOY_DIR for later deployment..."
+  popd &> /dev/null
+  else
```
Comment (davidcurrie) [inline source2image/assemble]: outdent else

File: source2image/assemble [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+  echo "Copying binaries in source directory into $DEPLOY_DIR for later deployment..."
+  fi
```
Comment (davidcurrie) [inline source2image/assemble]: outdent fi

File: source2image/assemble [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+  if [ -d $LOCAL_SOURCE_DIR/target ]; then
```
Comment (davidcurrie) [inline source2image/assemble]: outdent if/fi

File: source2image/assemble [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+  cp $LOCAL_SOURCE_DIR/target/*.war $DEPLOY_DIR >& /dev/null
+  chgrp -R 0 $DEPLOY_DIR
+  chmod -R g+rw $DEPLOY_DIR
+  fi
+
+  if [ -d $LOCAL_SOURCE_DIR/deployments ]; then
```
Comment (davidcurrie) [inline source2image/assemble]: outdent if

File: source2image/test/tests2i.sh [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+#Run the newly created image
+docker run -d -P --name libertys2i libertys2i
+
+#Test if the application has been deployed
+echo "Check that WebSphere Liberty has been created and that the application has been deployed"
+count=${2:-1}
+end=$((SECONDS+60))
+found=1
+	while (( $found != 0 && $SECONDS < $end ))
```
Comment (davidcurrie) [inline source2image/test/tests2i.sh]: reformat while loop

**Comment by davidcurrie** [comment]
Please address comments and then squash commits.

**Comment by jamiecoleman92** [comment]
I am having issue with my git history so I am putting these changes into a new repo in one commit rather than over 30 commits.

**Pull request closed without merging**